plugin/pkg/admission/nodeenv/admission_test.go

```diff
@@ -0,0 +1,198 @@
+	"k8s.io/kubernetes/pkg/api"
+	clientsetfake "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// TestPodAdmission verifies various scenarios involving pod/namespace/global node label selectors
+func TestPodAdmission(t *testing.T) {
+	namespace := &api.Namespace{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "testNamespace",
+			Namespace: "",
+		},
+	}
+
+	mockClientset := clientsetfake.NewSimpleClientset(namespace)
+	handler := &podNodeEnvironment{client: mockClientset}
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{Name: "testPod"},
+	}
+
+	tests := []struct {
+		namespaceNodeSelector           string
+		podNodeSelector                 map[string]string
+		mergedNodeSelector              map[string]string
+		ignoreTestNamespaceNodeSelector bool
+		admit                           bool
+		testName                        string
+	}{
+		{
+			podNodeSelector:                 map[string]string{},
+			mergedNodeSelector:              map[string]string{},
+			ignoreTestNamespaceNodeSelector: true,
+			admit:    true,
+			testName: "No node selectors",
+		},
+		{
+			namespaceNodeSelector: "infra=false",
+			podNodeSelector:       map[string]string{},
+			mergedNodeSelector:    map[string]string{"infra": "false"},
+			admit:                 true,
+			testName:              "Default node selector and no conflicts",
+		},
+		{
+			namespaceNodeSelector: " infra = false ",
+			podNodeSelector:       map[string]string{},
+			mergedNodeSelector:    map[string]string{"infra": "false"},
+			admit:                 true,
+			testName:              "TestNamespace node selector with whitespaces and no conflicts",
+		},
+		{
+			namespaceNodeSelector: "infra=false",
+			podNodeSelector:       map[string]string{},
+			mergedNodeSelector:    map[string]string{"infra": "false"},
+			admit:                 true,
+			testName:              "TestNamespace node selector and no conflicts",
+		},
+		{
+			namespaceNodeSelector: "",
+			podNodeSelector:       map[string]string{},
+			mergedNodeSelector:    map[string]string{},
+			admit:                 true,
+			testName:              "Empty namespace node selector and no conflicts",
+		},
+		{
+			namespaceNodeSelector: "infra=true",
+			podNodeSelector:       map[string]string{},
+			mergedNodeSelector:    map[string]string{"infra": "true"},
+			admit:                 true,
+			testName:              "Default and namespace node selector, no conflicts",
+		},
+		{
+			namespaceNodeSelector: "infra=true",
+			podNodeSelector:       map[string]string{"env": "test"},
+			mergedNodeSelector:    map[string]string{"infra": "true", "env": "test"},
+			admit:                 true,
+			testName:              "TestNamespace and pod node selector, no conflicts",
+		},
+		{
+			namespaceNodeSelector: "infra=true",
+			podNodeSelector:       map[string]string{"infra": "false"},
+			mergedNodeSelector:    map[string]string{"infra": "false"},
```

Comment:
I would omit this since admit=false - it's a bit confusing to see it here (and add a `continue` after line 111 if !test.admit)